### PR TITLE
Doc v2 h

### PR DESCRIPTION
**Pull request by jtlap**

### CODE REVIEW

File: include/eve/module/core/regular/hi.hpp [hunk 1]
```diff
@@ -48,14 +48,13 @@ namespace eve
 //!   @code
 //!   namespace eve
 //!   {
-//!      template< eve::value T >
-//!      auto hi(T x) noexcept;
+//!      constexpr auto hi(value auto x)                           noexcept;
```
Comment (jfalcou) [inline include/eve/module/core/regular/hi.hpp:51]: no masking ? We do support it normally now

File: include/eve/module/math/regular/horner.hpp [hunk 2]
```diff
@@ -53,12 +49,12 @@ namespace eve
...
-//!     T horner(T x, C ... coefs) noexcept;                                   //1
-//!
-//!     template< eve::floating_ordered_value T, eve::Range R>
-//!     T horner(T x, R r) noexcept;                                           //2
+//!      // Regular overloads
+//!      constexpr auto horner(value auto x, value auto ... cs)                      noexcept; // 1
+//!      constexpr auto horner(value auto x, kumi::non_empty_product_type auto tup)  noexcept; // 2
 //!
+//!      // Semantic options
+//!      constexpr auto horner[pedantic](/*any of the above overloads*/)             noexcept; // 3
```
Comment (jfalcou) [inline include/eve/module/math/regular/horner.hpp:57]: No masking ?

File: include/eve/module/core/regular/hi.hpp (hunk 1 above)
Comment (jtlap) [reply include/eve/module/core/regular/hi.hpp:51]: the return type does not allow that

File: include/eve/module/math/regular/horner.hpp (hunk 2 above)
Comment (jtlap) [reply include/eve/module/math/regular/horner.hpp:57]: horner was not masked

**Pull request merged**